test/puppetlabs/puppetdb/http/catalogs_test.clj

```diff
@@ -141,6 +141,48 @@
           response-body  (json/parse-string body true)]
       (is (= "myhost.localdomain" (:certname response-body))))))
 
+(deftestseq catalog-subqueries
+  [[version endpoint] endpoints
+   method [:get :post]]
+
+  (testcat/replace-catalog (json/generate-string catalog1))
+  (testcat/replace-catalog (json/generate-string catalog2))
+
+  (are [query expected]
+      (is (= (-> (reader (:body (query-response method endpoint query)))
+                 (json/parse-stream true)
+                 strip-hash
+                 set)
```

Comment:
sorry, that should just be the regular query-result. strip-hash can be passed as an optional handler.